syft/frameworks/torch/tensors/interpreters/chinese_remainder_theorem.py

```diff
@@ -0,0 +1,155 @@
+            assert f == r.child.field
+            self.child[f] = r
+            prod_modulo *= f
+            assert r.shape == res_shape, "All residue tensors of CRTTensor must have the same shape"
+
+    def __str__(self):
+        type_name = type(self).__name__
+        out = f"[" f"{type_name}]"
+        for mod, res in self.child.items():
+            out += f"\n\t {mod} -> " + str(res)
+        return out
+
+    def __repr__(self):
+        return self.__str__()
+    
+    @property
+    def shape(self) -> torch.Size:
+        return next(iter(self.child.values())).shape
+
+    @overloaded.method
+    def add(self, self_, other):
+        assert self_.keys() == other.keys(), "Cannot add 2 CRT that don't have the same modulos"
+
+        result = {}
+
+        for mod in self_.keys():
+            result[mod] = self_[mod] + other[mod]
+
+        return result
+
+    __add__ = add
+
+    @overloaded.method
+    def sub(self, self_, other):
+        assert (
+            self_.keys() == other.keys()
+        ), "Cannot subtract 2 CRT that don't have the same modulos"
+
+        result = {}
+
+        for mod in self_.keys():
+            result[mod] = self_[mod] - other[mod]
+
+        return result
+
+    __sub__ = sub
+
+    @overloaded.method
+    def mul(self, self_, other):
+        assert (
+            self_.keys() == other.keys()
+        ), "Cannot multiply 2 CRT that don't have the same modulos"
+
+        result = {}
+
+        for mod in self_.keys():
+            result[mod] = self_[mod] * other[mod]
+
+        return result
+
+    __mul__ = mul
+
+    def solve_system(self):
+        """ Build the tensor in Zq with q = prod(self.child.keys())
+        satisfying the modular system represented by the tensor.
+        The algorithm consists in:
+        1) Compute N = prod(self.child.keys())
+        2) yi = N / ni where the ni's are modulos in the system
+        3) zi = yi^(-1) mod ni (we know zi exists because all the original modulos are pairwise coprime
+        4) The result is x = sum(ai * yi * zi)
+        """
+        N = 1
+        for mod in self.child.keys():
+            N *= mod
+
+        def modular_inverse(a, b):
+            """ Computes the modular inverse x = a^(-1) mod b
+            with Euclid's extended algorithm.
+            """
+            b0 = b
+            x0, x1 = 0, 1
+            if b == 1: return 1
+            while a > 1:
+                q = a // b
+                a = b
+                b = a%b
+                x0 = x1 - q * x0
+                x1 = x0
+            if x1 < 0: x1 += b0
+            return x1
+    
+        res = 0
+        for mod, ai in self.child.items():
+            ai = ai.float_prec() % mod
+            yi = N // mod
+            zi = modular_inverse(yi, mod)
+            res += ai * yi * zi
+
+        return FixedPrecisionTensor(field=N, precision_fractional=0).on(res % N)
+
+    @staticmethod
+    @overloaded.module
+    def torch(module):
```

Comment:
what does this do?

Reply:
I wanted to allow the usage of torch.operation(a, b) with CRT tensors

Reply:
This is standard for overloading torch functions for special syft tensors